lib/generators/shopify_app/install/install_generator.rb

```diff
@@ -6,26 +6,19 @@ module Generators
     class InstallGenerator < Rails::Generators::Base
       include Rails::Generators::Migration
       source_root File.expand_path('../templates', __FILE__)
-      attr_reader :opts
 
-      def initialize(args, *options)
-        @opts = Hash[options.first.join(' ').scan(/--?([^=\s]+)(?:=(\S+))?/)]
-        @opts = @opts.with_indifferent_access
-        @opts.reverse_merge!(defaults)
-        super(args, *options)
-      end
-
-      def defaults
-        {
-          api_key: '<api_key>',
-          secret: '<secret>',
-          redirect_uri: 'http://localhost:3000/auth/shopify/callback',
-          scope: 'read_orders, read_products',
-          embedded: 'true'
-        }
-      end
+      class_option :api_key, type: :string, default: '<api_key>'
+      class_option :secret, type: :string, default: '<secret>'
+      class_option :redirect_uri, type: :string, default: 'http://localhost:3000/auth/shopify/callback'
+      class_option :scope, type: :string, default: 'read_orders, read_products'
+      class_option :embedded, type: :string, default: 'true'
 
       def create_shopify_app_initializer
+        @api_key = options['api_key']
```

Comment:
its part of the base generator class - the arguments defined in the class end up here